python/datafusion/dataframe.py

```diff
@@ -160,6 +160,40 @@ def with_column(self, name: str, expr: Expr) -> DataFrame:
         """
         return DataFrame(self.df.with_column(name, expr.expr))
 
+    def with_columns(
+        self, *exprs: Expr | Iterable[Expr], **named_exprs: Expr
+    ) -> DataFrame:
+        """Add an additional column to the DataFrame.
+
+        Args:
+            *exprs: Name of the column to add.
+            **named_exprs: Expression to compute the column.
```

Comment:
By reading this docstring I didn't at first understand the correct usage. I think I'd go with something more along the lines of

Add columns to the DataFrame.

The user can pass expressions, iteratables of expressions, or named expressions. To pass named expressions use the form `name=Expr`. For example the following will add 4 columns labeled `a`, `b`, `c`, and `d`.

```
df = df.with_columns(lit(0).alias('a'), [lit(1).alias('b'), lit(2).alias('c')], d=lit(3))
```

Reply:
I'll add an example later in the doc!